homeassistant/components/ambient_station/binary_sensor.py

```diff
@@ -37,23 +173,16 @@ async def async_setup_entry(
     """Set up Ambient PWS binary sensors based on a config entry."""
     ambient = hass.data[DOMAIN][DATA_CLIENT][entry.entry_id]
 
-    binary_sensor_list = []
-    for mac_address, station in ambient.stations.items():
-        for condition in station[ATTR_MONITORED_CONDITIONS]:
-            name, _, kind, device_class = SENSOR_TYPES[condition]
-            if kind == BINARY_SENSOR:
-                binary_sensor_list.append(
-                    AmbientWeatherBinarySensor(
-                        ambient,
-                        mac_address,
-                        station[ATTR_NAME],
-                        condition,
-                        name,
-                        device_class,
-                    )
-                )
-
-    async_add_entities(binary_sensor_list)
+    async_add_entities(
+        [
+            AmbientWeatherBinarySensor(
+                ambient, mac_address, station[ATTR_NAME], description
+            )
+            for description in BINARY_SENSOR_DESCRIPTIONS
+            for mac_address, station in ambient.stations.items()
```

Comment:
It likely won't change much here, but I would recommend to switch the iteration order.
Personally, I would write: For each station add these sensors. Not the other way round.